src/xinput.cc

```diff
@@ -46,30 +46,51 @@ int XInputTouch::find_touch(std::pair<XID,std::string> &ret)
 
 {
     XDeviceInfo	*devices;
-    int     found = -1;
+    std::pair<XID,std::string> ret_ts, ret_cal;
+    int         found_touchscreen = 0;
+    int         found_calib_matrix = 0;
     int		loop;
     int		num_devices;
 
     devices = XListInputDevices(display, &num_devices);
 
     for (loop=0; loop<num_devices; loop++) {
 
-        if (devices[loop].type != xi_touchscreen)
+        if (!devices[loop].type)
             continue;
```

Comment:
I guess so? My thought was that a device identifying as 'xi_touchscreen' should take priority over one identifying as a mouse with a calibration matrix, not breaking the functionality for any existing users who relied on this check.

If that's not important then I'd suggest we remove the check for xi_touchscreen completely, because without a "libinput Calibration Matrix", the code can't do anything useful with a device anyway.

Reply:
Yes you are right; I didn't read enough carefully your code: if there is a xi_touchscreen, it is used as target device, without considering any device with "libinput Calibration Matrix".

Anyway I am looking to understand when a "Calibration Matrix" property is attached to a device. This to better understand if we are facing a bug of a libinput/hid-driver or a real case.